src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java

```diff
@@ -0,0 +1,906 @@
+            else
+            {
+                numberOfEdges++;
+            }
+        }
+        final int numberOfAreas = changeSet.getLinesThatBecomeAreas().size();
+        final long numberOfLines = this.rawAtlas.numberOfLines()
+                - (numberOfAreas + changeSet.getCreatedEdges().size());
+
+        return new AtlasSize(numberOfEdges, changeSet.getPointsThatBecomeNodes().size(),
+                numberOfAreas, numberOfLines, changeSet.getPointsThatStayPoints().size(),
+                this.rawAtlas.numberOfRelations());
+    }
+
+    /**
+     * This function distinguishes between raw atlas points that will become {@link Point}s in the
+     * final {@link Atlas} or be simple shape points. We also explicitly call out the case where a
+     * raw atlas point may qualify to be an atlas {@link Node} by its tagging, but is not part of an
+     * {@link Edge}. In order to avoid creating floating {@link Node}s, we leave them as
+     * {@link Point}s in the final atlas and log the occurrence.
+     *
+     * @param changeSet
+     *            The {@link WaySectionChangeSet} to track any updates
+     */
+    private void distinguishPointsFromShapePoints(final WaySectionChangeSet changeSet)
+    {
+        this.rawAtlas.points().forEach(point ->
+        {
+            if (shouldSectionAtPoint(point)
+                    && !changeSet.getPointsThatBecomeNodes().contains(point.getIdentifier()))
+            {
+                logger.error("Point {} qualifies to be a Node, but isn't part of an Edge",
+                        point.getIdentifier());
+                changeSet.recordPoint(point);
+            }
+            else if (isAtlasPoint(changeSet, point))
+            {
+                changeSet.recordPoint(point);
+            }
+            else
+            {
+                // If a raw atlas point doesn't become a node or point in the way-sectioned atlas,
+                // it's a shape point and will be tracked by the underlying polyline of the area,
+                // line or edge that it's a part of. We can safely ignore handling them here.
+            }
+        });
+    }
+
+    /**
+     * This function takes care of identifying any raw atlas {@link Line}s that will becomes
+     * {@link Edge}s or {@link Area}s. If we happen to find an {@link Edge}, then we will loop
+     * through its shape points and identify all the {@link Node}s as well. The
+     * {@link WaySectionChangeSet} will be used to track of any entity updates. It's important to
+     * note that can avoid any future spatial queries by doing them here and creating a mapping to
+     * use later for splitting {@link Line}s into {@link Edge}s.
+     *
+     * @param changeSet
+     *            The {@link WaySectionChangeSet} to track any updates
+     */
+    private void identifyEdgesNodesAndAreasFromLines(final WaySectionChangeSet changeSet)
+    {
+        this.rawAtlas.lines().forEach(line ->
+        {
+            if (isAtlasEdge(line))
```

Comment:
stupid question: are there any atlas lines that might be both an edge and an area? It looks like we always prioritize edge in that case.

Reply:
Good question. I kept parity with the old slicing logic to have a 1:1 relationship between OSM Way to Atlas Edge/Line/Area. If this needs to be reconsidered, we can create an issue and it would be a pretty straightforward change.